scripts/build

```diff
@@ -61,15 +61,9 @@ do
 done
 
 # Checking for dependencies
-# XXX: we currently cannot check for the exact version of ic-cdk-optimizer
-# because of https://github.com/dfinity/cdk-rs/issues/181
-# Once the issue is fixed, we can ensure that the correct version is installed
-if ! command -v ic-cdk-optimizer
+if [[ ! "$(command -v ic-cdk-optimizer)" || "$(ic-cdk-optimizer --version)" != "ic-cdk-optimizer 0.3.4" ]]
 then
-    echo could not find ic-cdk-optimizer
-    echo "ic-cdk-optimizer version 0.3.1 is needed, please run the following command:"
-    echo "  cargo install ic-cdk-optimizer --version 0.3.1"
-    exit 1
+    cargo install ic-cdk-optimizer --version 0.3.4
```

Comment:
This will install the tool globally for the user, let's just let them know how they can install it instead

Reply:
I thought the build script is intended to be used inside a Docker container and not directly on the user's host machine.

Reply:
The build script is first and foremost meant to be called from by `dfx` and hence also on the user's host machine

Reply:
The README file already points to using Docker for the build and this could become the recommended way of running the build.

Reply:
I'm not installing ic-cdk-optimizer in the user build script anymore.